### DISCUSSION TITLE
Error on trying to disable/remove all snippets

### DISCUSSION THREAD

**Discussion by mikededo** [LazyVim/LazyVim#2539] [Q&A] [answered]
Hi, I'm trying to remove all snippets as I never use them and I would rather not have any snippet plugin installed, so that I can keep a cleaner config.

I have removed the `saadparwaiz1/cmp_luasnip` and `L3MON4D3/LuaSnip`, and still I'm getting the following error, which seems that there's something from the

<img width="1230" alt="image" src="https://github.com/LazyVim/LazyVim/assets/48334745/59e5be04-6630-4096-8f1b-6291266dac23">

Is there something I could do to avoid this error popping? I cannot tell in advance when the error will happen...

**Comment by dpetka2001** [comment, marked as answer]
You can't disable Luasnip itself. As stated in the `nvim-cmp` docs
>cmp.setup({
    snippet = {
      -- REQUIRED - you must specify a snippet engine
      expand = function(args)

You have to specify a snippet engine for it to work. You could disable the luasnip source for `nvim-cmp`, so you don't get snippets when the auto-completion popup menu opens with something like this
```lua
-- ~/.config/nvim/lua/plugins/snip.lua
return {
  {
    "nvim-cmp",
    opts = function(_, opts)
      -- local cmp = require("cmp")
      opts.sources = vim.tbl_filter(function(v)
        return not vim.tbl_contains({ "luasnip" }, v.name)
      end, opts.sources)
    end,
  },
}
```
You could also disable the plugin `rafamadriz/friendly-snippets` that LazyVim uses to provide snippets, if you want one more less plugin installed, since you won't be using it. But you should let `Luasnip` enabled if you want `nvim-cmp` to work.

**Reply by mikededo** [reply]
Understood, thanks. I missed `cmp` docs.
